2.2-taptweak.ipynb

```diff
@@ -146,6 +146,13 @@
     "# Method: sign_musig(private_key, nonce_key, nonce_point, public_key, msg)\n",
     "# Method: aggregate_musig_signatures(partial_signature_list, aggregate nonce)\n",
     "e = musig_digest(R_agg, agg_pubkey_tweaked, msg)\n",
+    "e_tweak = e * tweak\n",
+    "# Unlike `sign_musig` which takes care of negating the key if the publickey isn't quadratic residue\n",
```

Comment:
I find this a bit confusing. I think we need to negate the keys when aggregating them for musig (ie in `generate_musig_key()`). If we don't do that then the aggregate pubkey won't necessarily have a Y coordinate with a quadratic residue.

Reply:
Yes, I was debating what's the best way to handle musig and negations.
I thought musig is already pretty bloated so I can make `generate_musig_key`  take care of it by negating the musig coefficients instead and `sign_musig` checks and negates accordingly again which worked actually amazingly.

But then the tweak chapter came and ruin everything lol, so I'm not sure what's the best way forward, should we add manual privkey negations all over musig?